### DISCUSSION TITLE
The robot doesn't follow the curve trajectory smoothly

### DISCUSSION THREAD

**Discussion by wwookE** [visose/Robots#196] [Q&A] [answered]
Hi, I'm using Robots plug-in for my abb 2600 robot.

Now I'm working on simulating on actual robot with the code I extracted from Robots plug-in(by using 'SaveProgram' component)

The problem is, whenever I try to simulate with the curve target, the robot moves smoothly on grasshopper simulation, but 
when I run the same code with my actual robot, it moves discontinuously at each plane that I set on the curve.

Could anyone give me some advice on how to make the actual robot move smoothly?  

I attach files of grasshopper algorithm that I made from Robots, the RAPID code extracted from Robots. 
[RobotInquiry.zip](https://github.com/visose/Robots/files/10001038/RobotInquiry.zip)

**Comment by visose** [comment, marked as answer]
Your targets all have a Zone of 0 (aka 'fine' in ABB), which will cause the TCP to decelerate to a stop on every target.
Set a Zone input with a larger value. You can read about Zones in your ABB manual.

**Reply by wwookE** [reply]
The robot works well with your advice! 

I should also read about the Zones in my ABB manual as well

I really appreciate your help
